gateway-provider-security-hadoopauth/src/main/java/org/apache/knox/gateway/hadoopauth/filter/HadoopAuthFilter.java

```diff
@@ -201,6 +246,86 @@ public Principal getUserPrincipal() {
     super.doFilter(filterChain, request, response);
   }
 
+  /**
+   * A helper method that checks whether request contains
+   * unauthenticated path
+   * @param request
+   * @return
+   */
+  private boolean doesRequestContainUnauthPath(final ServletRequest request) {
+    for (final String path : unAuthenticatedPaths) {
```

Comment:
Is there a reason to not use `return unAuthenticatedPaths.contains(((HttpServletRequest) request).getPathInfo())` ?

Reply:
yup, that will simplify the code.